subsys/nrf_security/src/CMakeLists.txt

```diff
@@ -195,7 +191,7 @@ target_compile_options(${mbedcrypto_target}
 )
 
 # Add PSA core
-if(CONFIG_MBEDTLS_PSA_CRYPTO_C)
+if(CONFIG_MBEDTLS_PSA_CRYPTO_C AND NOT CONFIG_PSA_CORE_DISABLED)
```

Comment:
This is slightly confusing. Why have `CONFIG_MBEDTLS_PSA_CRYPTO_C` enabled in the first place?

Reply:
Multiple projects use `select MBEDTLS_PSA_CRYPTO_C`

Reply:
Why not change them to `imply` or add some condition to the `select`?

Reply:
That can be considered outside of this PR